### PR TITLE
Add delete (when files deleted in Explorer)

### PR DESCRIPTION
**Pull request by jeffyoung**
Also update telemetry (we don't want to send any on file operations)
Miscellaneous clean up

### CODE REVIEW

File: src/tfvc/commands/delete.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public GetOptions(): any {
+        return {};
+    }
+
+    //Delete returns either 0 (success) or 100 (failure).  IF we fail, simply throw.
+    public async ParseOutput(executionResult: IExecutionResult): Promise<string[]> {
+        let lines: string[] = CommandHelper.SplitIntoLines(executionResult.stdout, false, true /*filterEmptyLines*/);
+
+        if (executionResult.exitCode === 100) {
+            CommandHelper.ProcessErrors(this.GetArguments().GetCommand(), executionResult, true);
```
Comment (jpricketMSFT) [inline src/tfvc/commands/delete.ts]: Seems like this should come first. No need to split the lines if this throws. The exit code check also seems unnecessary if you only expect 100 or 0.

File: src/tfvc/commands/delete.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public GetArguments(): IArgumentProvider {
+        return new ArgumentBuilder("delete", this._serverContext)
+            .AddAll(this._itemPaths);
+    }
+
+    public GetOptions(): any {
+        return {};
+    }
+
+    //Delete returns either 0 (success) or 100 (failure).  IF we fail, simply throw.
```
Comment (jpricketMSFT) [inline src/tfvc/commands/delete.ts]: some example output would be helpful for understanding the parsing.

File: src/tfvc/scm/model.ts [hunk 3]
```diff
@@ -175,4 +179,20 @@ export class Model {
...
         return result;
     }
+
+    //When files are deleted in the VS Code Explorer, they are marked as Deleted but as candidate changes
+    //So we are looking for those and then running the Delete command on each.
+    private async processDeletes(changes: IPendingChange[]): Promise<void> {
+        let deleteCandidatePaths: string[] = [];
+        for (let index: number = 0; index < changes.length; index++) {
+            let change: IPendingChange = changes[index];
+            if (change.isCandidate && GetStatuses(change.changeType).find((e) => e === Status.DELETE)) {
```
Comment (jpricketMSFT) [inline src/tfvc/scm/model.ts]: You can use change.HasStatus(Status.DELETE) here. I think it does the same thing.

Comment (jeffyoung) [inline src/tfvc/scm/model.ts]: I need a Resource first (and don't have one at this point).  Instead of creating one, I'll do this instead.

**Pull request merged**